packages/health-check/src/utils.js

```diff
@@ -132,6 +154,92 @@ function isPortalModuleEnabled(module) {
   return process.env.PORTAL_MODULES && process.env.PORTAL_MODULES.indexOf(module) !== -1;
 }
 
+// sleep is a helper method of sleeping for for given time. The input time is
+// expected in seconds
+async function sleep(seconds) {
+  return new Promise((r) => setTimeout(r, seconds * SECOND));
+}
+
+// skylinkHealthCheck checks if the skylink has reached full redundancy
+async function skylinkHealthCheck(skylink, numRetries = 30, authCookie, isLarge = false) {
+  // Get the health of the skylink
+  const response = await got(`http://${siaDockerContainerIP}:${defaultSiaPort}/skynet/health/skylink/${skylink}`, {
+    headers: { "User-Agent": "Sia-Agent", cookie: authCookie },
+  });
+  const healthData = getResponseContent(response);
+
+  // Check Basesectorredundancy first
+  if (healthData.basesectorredundancy !== defaultBaseSectorRedundancy && numRetries > 0) {
+    // Semi-smart sleep before retrying. Sleep longer if the redundancy is
+    // lower.
+    let sleepTime = defaultBaseSectorRedundancy - healthData.basesectorredundancy;
+    await sleep(sleepTime);
+    return skylinkHealthCheck(skylink, numRetries - 1, authCookie, isLarge);
+  }
+
+  // Check the Fanout redundancy if it is a large file
+  if (isLarge && healthData.fanoutredundancy != defaultFanoutRedundancy && numRetries > 0) {
+    // Semi-smart sleep before retrying. Sleep longer if the redundancy is
+    // lower.
+    let sleepTime = (defaultFanoutRedundancy - healthData.fanoutredundancy) * 10;
+    await sleep(sleepTime);
+    return skylinkHealthCheck(skylink, numRetries - 1, authCookie, isLarge);
+  }
+
+  // Throw error if the basesectorredundancy never reached 10x
+  if (healthData.basesectorredundancy !== defaultBaseSectorRedundancy && numRetries === 0) {
+    throw new Error(`File uploaded but basesector did not reach full redundancy: ${healthData.basesectorredundancy}`);
+  }
+
+  // Throw error if the fanoutredundancy never reached 3x
+  if (isLarge && healthData.fanoutredundancy !== defaultFanoutRedundancy && numRetries === 0) {
+    throw new Error(`File uploaded but fanout did not reach full redundancy: ${healthData.fanoutredundancy}`);
+  }
+
+  return response;
```

Comment:
just a suggestion, rewriting it this way makes it easy to ensure you've covered all possible outcomes

```javascript
const { basesectorredundancy, fanoutredundancy} = healthData
const baseSectorRedundancyReached = basesectorredundancy >= DEFAULT_BASE_SECTOR_REDUNDANCY
const fanoutSectorRedundancyReached = fanoutredundancy >= DEFAULT_FANOUT_REDUNDANCY
const sleepForInS = isLarge
    ? (DEFAULT_FANOUT_REDUNDANCY - fanoutredundancy) * 10;
    : (DEFAULT_BASE_SECTOR_REDUNDANCY - basesectorredundancy) * 10;

const healthy = baseSectorRedundancyReached && (!isLarge || fanoutSectorRedundancyReached)
if (!healthy && attemptsLeft > 0 ) {
    await sleep(sleepForInS)
    return skylinkHealthCheck(skylink, attemptsLeft - 1, authCookie, isLarge);
}

if (!baseSectorRedundancyReached) { throw new Error(...) }
if (!fanoutSectorRedundancyReached) { throw new Error(...) }
```